src/components/application_manager/test/policy_handler_test.cc

```diff
@@ -1467,24 +1668,919 @@ TEST_F(PolicyHandlerTest, SendMessageToSDK) {
+ * threads, or a timeout expires.
+ */
+class WaitAsync {
+ public:
+  WaitAsync(const uint32_t count, const uint32_t timeout)
+      : count_(count), timeout_(timeout) {}
+
+  void Notify() {
+    count_--;
+    cond_var_.NotifyOne();
+  }
+
+  bool Wait(sync_primitives::AutoLock& auto_lock) {
+    while (count_ > 0) {
+      sync_primitives::ConditionalVariable::WaitStatus wait_status =
+          cond_var_.WaitFor(auto_lock, timeout_);
+      if (wait_status == sync_primitives::ConditionalVariable::kTimeout) {
+        return false;
+      }
+    }
+    return true;
+  }
+
+ private:
+  int count_;
+  const uint32_t timeout_;
+  sync_primitives::ConditionalVariable cond_var_;
+};
+}
+
+ACTION_P(NotifyAsync, waiter) {
+  waiter->Notify();
+}
+
+TEST_F(PolicyHandlerTest,
+       OnAppPermissionConsentInternal_ValidConnectionKey_SUCCESS) {
+  // EnablePolicyAndPolicyManagerMock();
+
+  ChangePolicyManagerToMock();
+  const uint32_t device = 2u;
+
+  PermissionConsent permissions;
+  permissions.device_id = kDeviceId_;
+  permissions.consent_source = "consent_source";
+
+  policy::FunctionalGroupPermission group_permission_allowed;
+  CreateFunctionalGroupPermission(GroupConsent::kGroupAllowed,
+                                  "allowed",
+                                  "name_allowed",
+                                  group_permission_allowed);
+
+  permissions.group_permissions.push_back(group_permission_allowed);
+
+  EXPECT_CALL(app_manager_, connection_handler())
+      .WillOnce(ReturnRef(conn_handler));
+  EXPECT_CALL(conn_handler, get_session_observer())
+      .WillOnce(ReturnRef(mock_session_observer));
+  EXPECT_CALL(mock_session_observer, GetDataOnDeviceID(device, _, NULL, _, _))
+      .WillOnce(Return(1u));
+
+  EXPECT_CALL(app_manager_, application(kConnectionKey_))
+      .WillOnce(Return(mock_app_));
+  EXPECT_CALL(*mock_app_, policy_app_id()).WillOnce(Return(kPolicyAppId_));
+  EXPECT_CALL(*mock_app_, device()).WillOnce(Return(device));
+
+  sync_primitives::Lock wait_hmi_lock;
+  sync_primitives::AutoLock auto_lock(wait_hmi_lock);
+  const uint32_t calls_count = 1u;
+  const uint32_t timeout = 2u * 1000u;
+  WaitAsync waiter(calls_count, timeout);
+
+  EXPECT_CALL(*mock_policy_manager_, SetUserConsentForApp(_))
+      .WillOnce(NotifyAsync(&waiter));
+
+  policy_handler_.OnAppPermissionConsent(kConnectionKey_, permissions);
+  EXPECT_TRUE(waiter.Wait(auto_lock));
+}
+
+TEST_F(PolicyHandlerTest,
+       OnAppPermissionConsentInternal_NoAppsPreviouslyStored_UNUSUCCESS) {
+  ChangePolicyManagerToMock();
+  const uint32_t invalid_connection_key = 0u;
+
+  PermissionConsent permissions;
+  permissions.device_id = kDeviceId_;
+  permissions.consent_source = "consent_source";
+
+  policy::FunctionalGroupPermission group_permission_allowed;
+  CreateFunctionalGroupPermission(GroupConsent::kGroupAllowed,
+                                  "allowed",
+                                  "name_allowed",
```

Comment:
@okozlovlux, You use this kind of strings like a `"allowed"`, `"name_allowed"` and `"consent_source"`, in several test cases. May you move it to an nameless namespace or fixture class?